python/ray/tune/ray_trial_executor.py

```diff
@@ -212,16 +226,47 @@ def _setup_remote_runner(self, trial, reuse_allowed):
             logger.debug("Cannot reuse cached runner {} for new trial".format(
                 self._cached_actor))
             with self._change_working_directory(trial):
-                self._trial_cleanup.add(trial, actor=self._cached_actor)
+                self._trial_cleanup.add(
+                    trial,
+                    actor=self._cached_actor,
+                    placement_group=self._committed_placement_groups.get(
+                        trial, [None])[0])
+                if trial in self._committed_placement_groups:
+                    del self._committed_placement_groups[trial]
             self._cached_actor = None
 
         _actor_cls = _class_cache.get(trial.get_trainable_cls())
-        full_actor_class = _actor_cls.options(
-            num_cpus=trial.resources.cpu,
-            num_gpus=trial.resources.gpu,
-            memory=trial.resources.memory or None,
-            object_store_memory=trial.resources.object_store_memory or None,
-            resources=trial.resources.custom_resources)
+        if isinstance(trial.resources, PlacementGroupFactory):
+            if trial not in self._committed_placement_groups:
+                # This creates the placement group and will try to
+                # allocate resources
+                placement_group = trial.resources()
+                ready_fut = placement_group.ready()
+                self._committed_placement_groups[
+                    trial] = placement_group, ready_fut
+            else:
+                placement_group, ready_fut = self._committed_placement_groups[
+                    trial]
+            ready, _ = ray.wait([ready_fut], timeout=0)
+            if not ready:
+                return None
+            if not ray.get(ready):
+                # Schedule new future
+                self._committed_placement_groups[
+                    trial] = placement_group, placement_group.ready()
+                return None
+            # The trainable actor always lives on the first bundle
+            full_actor_class = _actor_cls.options(
+                placement_group=placement_group,
+                placement_group_bundle_index=0)
```

Comment:
i thought we still need to provide the resource requirements for this actor, even though we are targeting a specific bundle?

Reply:
Yeah placement group is not reflected if it doesn’t require resources

Reply:
So we do have a test in this PR that checks for `ray.available_resources()` after scheduling an actor with placement groups. It seems the available resources are correctly subtracted (see `testResourcesAvailableWithPlacementGroup`).

I was under the impression that specifying resources is only necessary if we want to auto-select a bundle. If we pass a bundle index, shouldn't the bundle specification imply which resources to request?

Reply:
Also, the docs page does not pass any resource requests in addition to the placement group and bundle id: https://docs.ray.io/en/master/placement-group.html